protocols/raft/src/main/java/io/atomix/protocols/raft/roles/LeaderAppender.java

```diff
@@ -135,16 +135,10 @@ private int getQuorumIndex() {
       return CompletableFuture.completedFuture(index);
     }
 
-    // If there are no other stateful servers in the cluster, immediately commit the index.
-    if (raft.getCluster().getActiveMemberStates().isEmpty() && raft.getCluster().getPassiveMemberStates().isEmpty()) {
-      long previousCommitIndex = raft.getCommitIndex();
-      raft.setCommitIndex(index);
-      completeCommits(previousCommitIndex, index);
-      return CompletableFuture.completedFuture(index);
-    }
+    // If there are no other stateful servers in the cluster, immediately commit the index OR
     // If there are no other active members in the cluster, update the commit index and complete the commit.
     // The updated commit index will be sent to passive/reserve members on heartbeats.
-    else if (raft.getCluster().getActiveMemberStates().isEmpty()) {
+    if (raft.getCluster().getActiveMemberStates().isEmpty()) {
```

Comment:
lol :see_no_evil:

Reply:
Yeah not related to this PR but I saw it and thought...wait a minute... :sweat_smile: